rest/rest-server/src/main/java/org/apache/brooklyn/rest/resources/ApplicationResource.java

```diff
@@ -266,27 +268,48 @@ protected Response createFromAppSpec(ApplicationSpec applicationSpec) {
     @Override
     public Response createFromYaml(String yaml) {
         // First of all, see if it's a URL
-        URI uri;
+        Preconditions.checkNotNull(yaml, "Blueprint cannot be null");
```

Comment:
Personal preference for "Blueprint must not be null", given that it is null. No strong feelings though. Happy for you to ignore this :-)